src/chain/chain.rs

```diff
@@ -540,15 +548,19 @@ impl Chain {
 
     /// Generate a new section info based on the current set of members.
     /// Returns a set of EldersInfos to vote for.
-    pub fn promote_and_demote_elders(&mut self) -> Result<Vec<EldersInfo>, RoutingError> {
-        self.assert_no_prefix_change("promote and demote");
+    fn promote_and_demote_elders(&mut self) -> Result<Option<Vec<EldersInfo>>, RoutingError> {
+        if !self.members_changed || !self.can_poll_churn() {
+            // Nothing changed that could impact elder set, or we cannot process it yet.
+            return Ok(None);
```

Comment:
Why not just returning an empty vec?

Reply:
Or maybe change the return type to something like:
```rust
enum NewEldersInfos {
    None,
    One(EldersInfo),
    Two(EldersInfo),
}
```
(but with better names)

Reply:
The empty vec make the call site/intent less clear (poll_accumulated), so in the end I converted it to an option.

The One, Tow would also make the call site (futher up) less clear as instead of looping through all the given info we'd have to have specific code for either case.

Reply:
You can `impl IntoIterator` for the `NewEldersinfos` thing if you want to be really fancy :) But I'm OK with it as is for now. We can always give it a refactoring pass some time later if we want.